R/install-github.R

```diff
@@ -235,12 +239,21 @@ github_resolve_ref.github_release <- function(x, params) {
 #' @export
 #' @examples
 #' parse_github_repo_spec("metacran/crandb")
-#' parse_github_repo_spec("jeroenooms/curl@v0.9.3")
+#' parse_github_repo_spec("jeroen/curl@v0.9.3")
 #' parse_github_repo_spec("jimhester/covr#47")
 #' parse_github_repo_spec("hadley/dplyr@*release")
 #' parse_github_repo_spec("mangothecat/remotes@550a3c7d3f9e1493a2ba")
+#' parse_github_repo_spec("https://github.com/jimhester/covr")
+#' parse_github_repo_spec("https://github.com/jeroen/curl.git")
+#' parse_github_repo_spec("git@github.com:metacran/crandb.git")
 
 parse_github_repo_spec <- function(repo) {
+
+  repo <- gsub("\\.git$", "", repo)
```

Comment:
I would only remove the `.git` suffix with the prefixes, because some git urls might not handle the `.git`-less form, like GitHub.

Reply:
OK I can do that. But just to check ... what are you actually worried about? That there might be a `.git` I should *not* strip? That there might be a GitHub repo actually named `foo.git` and `jenny/foo.git` should return username = `jenny` and repo = `foo.git`?